### PR TITLE
[DOC] Document `evaluate` output DataFrame schema 

### LINKED ISSUE #3698: [DOC] `pd.DataFrame` output schema of `evaluate` needs to be documented properly in docstring
The docstring of `evaluate` in the `forecasting.model_evaluation` module only tells the user that the return is `pd.DataFrame` without further description of the schema.

Best way to glean the schema is to run the example; alternatively, the schema is tested by calls to `_check_evaluate_output` in the tests.

### PR DESCRIPTION
**Pull request by topher-lo**
#### Reference Issues/PRs
Closes #3698

#### What does this implement/fix? Explain your changes.
Documents all columns in DataFrame of results returned by `evaluate`

#### Does your contribution introduce a new dependency? If yes, which one?
Np.

#### What should a reviewer concentrate their feedback on?
Docstring.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

### CODE REVIEW

File: sktime/forecasting/model_evaluation/_functions.py [hunk 1]
```diff
@@ -263,9 +263,18 @@ def evaluate(
 
     Returns
     -------
-    pd.DataFrame
+    results : pd.DataFrame or dask.dataframe.DataFrame
         DataFrame that contains several columns with information regarding each
         refit/update and prediction of the forecaster.
+        Returned DataFrame contains following columns:
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:269]: contains -> has

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks for the swift fix!

Can you kindly also describe what the rows are (incl row index)?

File: sktime/forecasting/model_evaluation/_functions.py [hunk 2]
```diff
@@ -263,9 +263,18 @@ def evaluate(
...
+        - test_{scoring.name}: (float) Model performance scores.
+        - fit_time: (float) Time in seconds to fit estimator on train window.
+        - pred_time: (float) Time in seconds to get forecasts from fitted estimator.
+        - len_train_window: (int) Length of train window.
+        - cutoff: (int, pd.Timestamp, pd.Period) Last time point of train window.
+        - y_train: (pd.Series) Optional; see `return_data`.
```
Comment (fkiraly) [inline sktime/forecasting/model_evaluation/_functions.py:275]: should also be described here

File: sktime/forecasting/model_evaluation/_functions.py (hunk 1 above)
Comment (topher-lo) [inline sktime/forecasting/model_evaluation/_functions.py:269]:
```suggestion
        Returned DataFrame has the following columns:
```

File: sktime/forecasting/model_evaluation/_functions.py (hunk 2 above)
Comment (topher-lo) [inline sktime/forecasting/model_evaluation/_functions.py:275]:
```suggestion
        - y_train: (pd.Series) Optional; see `return_data`. Training data.
```

Comment (topher-lo) [inline sktime/forecasting/model_evaluation/_functions.py:275]:
```suggestion
        - y_train: (pd.Series) Optional, see `return_data`; data used to train forecaster at n-th window.
```

File: sktime/forecasting/model_evaluation/_functions.py [hunk 3]
```diff
@@ -263,9 +263,18 @@ def evaluate(
...
         DataFrame that contains several columns with information regarding each
         refit/update and prediction of the forecaster.
+        Returned DataFrame has the following columns:
+        - test_{scoring.name}: (float) Model performance scores.
+        - fit_time: (float) Time in seconds to fit estimator on train window.
+        - pred_time: (float) Time in seconds to get forecasts from fitted estimator.
+        - len_train_window: (int) Length of train window.
+        - cutoff: (int, pd.Timestamp, pd.Period) Last time point of train window.
+        - y_train: (pd.Series) Optional, see `return_data`; data used to train forecaster at n-th window.
+        - y_pred: (pd.Series) Optional; see `return_data`.
```
Comment (topher-lo) [inline sktime/forecasting/model_evaluation/_functions.py:276]:
```suggestion
        - y_pred: (pd.Series) Optional, see `return_data`; forecasts from fitted forecaster at n-th window.
```

File: sktime/forecasting/model_evaluation/_functions.py [hunk 4]
```diff
@@ -263,9 +263,18 @@ def evaluate(
...
+        - y_test: (pd.Series) Optional; see `return_data`.
```
Comment (topher-lo) [inline sktime/forecasting/model_evaluation/_functions.py:277]:
```suggestion
        - y_test: (pd.Series) Optional, see `return_data`; testing data at n-th window.
```

File: sktime/forecasting/model_evaluation/_functions.py [hunk 5]
```diff
@@ -263,9 +263,18 @@ def evaluate(
 
     Returns
     -------
-    pd.DataFrame
+    results : pd.DataFrame or dask.dataframe.DataFrame
         DataFrame that contains several columns with information regarding each
         refit/update and prediction of the forecaster.
```
Comment (topher-lo) [inline sktime/forecasting/model_evaluation/_functions.py:268]:
```suggestion
        refit/update and prediction of the forecaster.
        The DataFrame is indexed by the i-th train-test window.
        A single row represents the metrics from the forecaster fitted on the i-th window.
```

**Pull request merged**